bpf/kmesh/workload/include/authz.h

```diff
@@ -32,79 +49,187 @@ static inline wl_policies_v *get_workload_policies_by_uid(__u32 workload_uid)
     return (wl_policies_v *)kmesh_map_lookup_elem(&map_of_wl_policy, &workload_uid);
 }
 
-static inline int matchDstPorts(Istio__Security__Match *match, struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
+static inline int parser_xdp_info(struct xdp_md *ctx, struct xdp_info *info)
+{
+    void *begin = (void *)(long)(ctx->data);
+    void *end = (void *)(long)(ctx->data_end);
+
+    // eth header
+    info->ethh = (struct ethhdr *)begin;
+    if ((void *)(info->ethh + 1) > end)
+        return PARSER_FAILED;
+
+    // ip4|ip6 header
+    begin = info->ethh + 1;
+    if ((begin + 1) > end)
+        return PARSER_FAILED;
+    if (((struct iphdr *)begin)->version == 4) {
+        info->iph = (struct iphdr *)begin;
+        if ((void *)(info->iph + 1) > end || (info->iph->protocol != IPPROTO_TCP))
+            return PARSER_FAILED;
+        begin = (info->iph + 1);
+    } else if (((struct iphdr *)begin)->version == 6) {
+        info->ip6h = (struct ipv6hdr *)begin;
+        if ((void *)(info->ip6h + 1) > end || (info->ip6h->nexthdr != IPPROTO_TCP))
+            return PARSER_FAILED;
+        begin = (info->ip6h + 1);
+    } else
+        return PARSER_FAILED;
+
+    info->tcph = (struct tcphdr *)begin;
+    if ((void *)(info->tcph + 1) > end)
+        return PARSER_FAILED;
+    return PARSER_SUCC;
+}
+
+static inline void parser_tuple(struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
 {
+    if (info->iph->version == 4) {
+        tuple_info->ipv4.saddr = info->iph->saddr;
+        tuple_info->ipv4.daddr = info->iph->daddr;
+        tuple_info->ipv4.sport = info->tcph->source;
+        tuple_info->ipv4.dport = info->tcph->dest;
+    } else {
+        bpf_memcpy((__u8 *)tuple_info->ipv6.saddr, info->ip6h->saddr.in6_u.u6_addr8, IPV6_ADDR_LEN);
+        bpf_memcpy((__u8 *)tuple_info->ipv6.daddr, info->ip6h->daddr.in6_u.u6_addr8, IPV6_ADDR_LEN);
+        tuple_info->ipv6.sport = info->tcph->source;
+        tuple_info->ipv6.dport = info->tcph->dest;
+    }
+}
+
+static inline int get_tuple_key(struct xdp_md *ctx, struct bpf_sock_tuple *tuple_info, struct xdp_info *info)
+{
+    int ret = parser_xdp_info(ctx, info);
+    if (ret != PARSER_SUCC) {
+        BPF_LOG(ERR, AUTH, "Failed to parse xdp_info\n");
+        return PARSER_FAILED;
+    }
+
+    parser_tuple(info, tuple_info);
+
+    return PARSER_SUCC;
+}
+
+SEC("xdp_auth")
+int matchDstPorts(struct xdp_md *ctx)
```

Comment:
obey c naming convention
```suggestion
int match_dst_port(struct xdp_md *ctx)
```

Reply:
how do you make sure this prog is not triggered each time a packet received